deploy/charts/cert-manager/templates/service.yaml

```diff
@@ -21,6 +21,9 @@ spec:
   type: ClusterIP
   ports:
   - protocol: TCP
+{{- if .Values.prometheus.servicemonitor.appProtocol }}
+    appProtocol: {{ .Values.prometheus.servicemonitor.appProtocol }}
+{{- end }}
     port: 9402
     name: tcp-prometheus-servicemonitor
```

Comment:
@jsievenpiper are you saying that this port should be named `http-prometheus-servicemonitor` ideally?
How will we break users by just renaming the port?

Reply:
For automatic protocol detection to work (in the case of something like Istio) -- yes, ideally it'd be renamed.

I think technically there are cases where you may have other manifests that point to this port by name (e.g. I configure another resource to look at this service port `tcp-prometheus-servicemonitor`). I'm not personally aware of any such resources, but I know they're pretty common for container port names. Specifying `appProtocol` works as well, and conforms to the K8S standard options, so I figured that was a fair compromise that's backwards and forwards compatible and doesn't bend the naming to any one particular vendor of a solution.